### ISSUE TITLE
AHardwarebuffer and rectangle comparing on GPU

### ISSUE THREAD

**Issue by bk138** [bk138/droidVNC-NG#3]

**Comment by wawahuy** [comment]
Do you still want to develop this feature? I think I can try it.

**Comment by bk138** [comment]
Yeah sure, otherwise it would be closed. Let me know if you need any pointers to documentation. It only makes sense together with #2 I think and is probably at least a week of work.

**Comment by wawahuy** [comment]
I have read the 'a-hardware-buffer', but I am unsure how to compare two images.  Perhaps I need to incorporate OpenCL. What are your thoughts on this?

**Comment by bk138** [comment]
It's maybe easier to go a bit more high-level and do the comparing in managed code, i.e. here https://github.com/bk138/droidVNC-NG/blob/master/app/src/main/java/net/christianbeier/droidvnc_ng/MainService.java#L622 - this probably involves converting the Image's to Bitmaps and then doing comparisons. But it's rather really #2 then so let's continue there!

**Comment by bk138** [comment]
Note to self: Android 16 and 16.1 and 17 x86 emulator images seem to have a memory leak wrt image.getPlanes() - fd count goes constantly up when MediaProjection is on, even with no clients connected. Cannot repro on Android 15 emulator nor on Android 16 hardware target (so likely an emu thing...).
